etl-job/fhir_import_export.py

```diff
@@ -184,40 +178,19 @@ def _download_and_unzip(object_id: str,
     return True
 
 
-def _load_all(study: str,
-              project_id: str,
+def _load_all(program: str,
+              project: str,
               output: dict,
               file_path: str,
-              schema: str,
               work_path: str) -> bool:
 
-    if study is None or study == "":
-        output['logs'].append("Please provide a study name")
-        return False
-
-    if project_id is None or project_id == "":
-        output['logs'].append("Please provide a project_id (program-project)")
-        return False
-
     logs = None
     try:
-        program, project = project_id.split('-')
-        assert program, output['logs'].append("program is required")
-        assert project, output['logs'].append("project is required")
-
-        file_path = pathlib.Path(file_path)
-        extraction_path = file_path / 'extractions'
-        research_study = str(extraction_path / 'ResearchStudy.ndjson')
-
-        file_path = str(file_path)
-        extraction_path = str(extraction_path)
-        output['logs'].append(f"Simplifying study: {file_path}")
-
-        # call jsonschemagraph to create edges and vertices
-        graph_gen_cmd = ["jsonschemagraph", "gen-dir", "iceberg/schemas/graph", f"{file_path}", f"{extraction_path}","--project_id", f"{project_id}","--gzip_files"]
-        subprocess.run(graph_gen_cmd, check=True, capture_output=True)
-
-        bulk_load(_get_grip_service(), "CALIPER",f"{program}-{project}", extraction_path, output, _get_token())
+        for file in pathlib.Path(file_path).rglob('*'):
+            if file.suffix in ['.ndjson', '.json']:
+                # output dictionary is capturing logs from this function
+                bulk_load_raw(_get_grip_service(), "CALIPER",
+                    f"{program}-{project}", str(file), output, _get_token())
```

Comment:
Adding your response from chatting...
> So before CLI was running this block: https://github.com/bmeg/jsonschemagraph/blob/main/cmd/gen_dir/main.go now it runs an http request this code block: https://github.com/bmeg/grip/blob/feature/fhir-json-schema/server/api.go#L225-L325. The main elements like jsonschemagraph.Generate and jsonschemagraph.Load are still there, but the Load function got modified to load from RAM instead of loading from a schema file: https://github.com/bmeg/grip/blob/feature/fhir-json-schema/server/metagraphs.go#L169-L196 and the generate is still being called as it is in the jsonschemagraph lib, it’s just wrapped in a bunch of channels in order to do a protobuf stream operation instead of a local CLI call like it had been done before. https://github.com/bmeg/grip/blob/feature/fhir-json-schema/server/api.go#L289-L320

Got it, so all generation and validation is happening in GRIP itself, right when the vertices are being generated. And so was that the purpose of moving the validation: to house all of the graph-related checks within GRIP where the actual graph is being processed?

Reply:
before the data was downloaded from bucket, unzipped, the CLI is used to do edge generation and these edges were then saved to disk and then loaded from disk to do an HTTP request to grip.

now the data is downloaded from  bucket, unzipped, and passed directly to grip via an HTTP saving an entire read and write of the data and saving a read and a download of the schema from disk on every single job. 

Now the schema is added to grip when the grip server turns on as a init-db-job and that same schema is used as long as grip is running, removing the need to download the schema on every job.